### PR TITLE
cleanup: remove istio_authn generator

### PR DESCRIPTION
**Pull request by kyessenov**
Change-Id: Ibd951922e020110cc5f7cd25bfa10202c7b9bbbc

The code for istio_authn has been gone since 1.21.

### CODE REVIEW

File: pilot/pkg/security/authz/builder/BUILD [hunk 1]
```diff
@@ -0,0 +1,17 @@
+go_library(
```
Comment (howardjohn) [inline pilot/pkg/security/authz/builder/BUILD:1]: can you remove this?

**Review by howardjohn — APPROVED** [review]
LGTM, put a hold for the bazel file though

File: pilot/pkg/security/authz/builder/BUILD (hunk 1 above)
Comment (zirain) [reply pilot/pkg/security/authz/builder/BUILD:1]: google use Bazel to buid go internally?

Comment (kyessenov) [reply pilot/pkg/security/authz/builder/BUILD:1]: oops, yeah, sorry. not sure why it was generated, I didn't do anything...

@zirain it does, but not for OSS code.

Comment (zirain) [reply pilot/pkg/security/authz/builder/BUILD]: good to know that, thanks!

**Pull request merged**